aeneas/src/ssa/SsaOptimizer.v3

```diff
@@ -835,6 +835,21 @@ class SsaInstrReducer(context: SsaContext) extends SsaInstrMatcher {
 					return replaceOp(i, V3Op.newClassGetMethod(dv));  // ClassGetVirtual[m](K) => ClassGetMethod[m](K)
 				}
 			}
+			VariantGetField(field) => {
+				var receiver = i.input0();
+				if (state.end) return receiver;
+				var unboxed = ClassType.!(receiver.getType()).classDecl.isUnboxed();
+				if (SsaConst.?(receiver)) {
+					// ClassGetField(#K) => #K
+					return graph.valConst(getFieldType(i.op, field),
+						asRecord(receiver).values[field.index]);
+				}
+				i.setFactIf(Fact.O_NO_NULL_CHECK, Fact.O_PURE);
+				if (!unboxed) {
```

Comment:
I think the `F_VALUE` isn't necessary here; it should apply to a field declaration and isn't meant for SSA instructions.